bin/embedding-sdk/fixup-types-after-compilation.js

```diff
@@ -69,21 +65,120 @@ const replaceAliasedImports = filePath => {
   });
 
   fs.writeFileSync(filePath, fileContent, { encoding: "utf-8" });
-  console.log(`Edited file: ${filePath}`);
+  log(`Edited file: ${filePath}`);
+};
+
+const removeUnresolvedReexports = (filePath) => {
+  if (!filePath.endsWith("index.d.ts")) {
+    return;
+  }
+
+  const fileContent = fs.readFileSync(filePath, { encoding: "utf8" });
+  const lines = fileContent.split(/\r?\n/);
+
+  let isModified = false;
+
+  const updatedContent = lines
+    .filter((line) => {
+      if (!line.includes("export * from")) {
+        return true;
+      }
+
+      const target = line.match(/export \* from "(.*)"/)[1];
+      const dirPath = path.dirname(filePath);
+
+      const isUnresolved =
+        !fs.existsSync(path.resolve(dirPath, target)) &&
+        !fs.existsSync(`${path.resolve(dirPath, target)}.d.ts`);
+
+      if (isUnresolved) {
+        log(`Removing unresolved re-export: ${line}`);
+        isModified = true;
+      }
+
+      return !isUnresolved;
+    })
+    .join("\n");
+
+  if (isModified) {
+    fs.writeFileSync(filePath, updatedContent, { encoding: "utf-8" });
+  }
 };
 
-const fixupTypesAfterCompilation = () => {
-  console.log("[dts fixup] Fixing SDK d.ts files...");
+const fixupTypesAfterCompilation = ({ isWatchMode }) => {
+  log("Fixing SDK d.ts files...");
+
+  const dtsFilePaths = glob.sync(`${SDK_DIST_DIR_PATH}/**/*.d.ts`);
+
+  dtsFilePaths.forEach((filePath) => {
+    replaceAliasedImports(filePath);
+    removeUnresolvedReexports(filePath);
+  });
 
-  const dtsFilePaths = traverseFilesTree(SDK_DIST_DIR_PATH);
+  log("Done!");
 
-  dtsFilePaths.forEach(replaceAliasedImports);
+  if (!isWatchMode) {
+    generateDtsRollup();
+  }
+};
+
+const generateDtsRollup = () => {
+  // Dts rollup logger
+  const { log, error } = getLogger("dts rollup");
+
+  log("Generate dts rollup...");
+
+  const dtsRollupEntryPointPath = path.resolve(
+    path.join(
+      __dirname,
+      "../../resources/embedding-sdk/dist/enterprise/frontend/src/embedding-sdk/index.d.ts",
+    ),
+  );
+  const dtsRollupEntryPointPathExist = fs.existsSync(dtsRollupEntryPointPath);
+
+  if (!dtsRollupEntryPointPathExist) {
+    log("It looks like dts rollup is already generated, skipping...");
+
+    return;
+  }
```

Comment:
Could the DTS rollup path be stale? (e.g. if it was already generated but out-of-date)

Reply:
Dts rollup should always happen after the regular (granular) dts files are generated
And during dts rollup, we clean up these files, keeping only the main `index.d.ts` file.

Dts rollup uses one of the granular `d.ts` files as the entry point (that we check in this condition), so if we run dts rollup twice without regenerating granular `.d.ts` files, it will throw an error, because for the second run the entry point `.d.ts` file is removed, and this condition handles this case.

The expected flow is to always run it as the part of the `yarn build-embedding-sdk` command

Reply:
Ah right, assuming that the cleanup runs successfully this should be fine 👍🏻